src/components/molecules/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,94 @@
+import { isActiveFeatureName } from '@paralleldrive/feature-toggles'
+import { Feature } from '@paralleldrive/react-feature-toggles'
+
+import * as classNames from 'classnames'
+import * as React from 'react'
+import { browserHistory } from 'react-router'
+import { Action } from 'redux'
+
+import { AvatarMenu } from 'components/atoms/AvatarMenu/AvatarMenu'
+import { NavButtons } from 'components/molecules/NavButtons/NavButtons'
+import { FeatureName } from 'config/features'
+import { Images } from 'images/Images'
+import { User } from 'interfaces/Props'
+
+import './Navbar.scss'
+
+interface NavbarActions {
+  readonly onSearchChange?: (searchQuery: string) => Action
+  readonly onSignOut?: () => Action
+}
+
+export interface NavbarProps {
+  readonly shadow?: boolean
+  readonly transparent?: boolean
+  readonly margin?: boolean
+  readonly displayLogo?: boolean
+  readonly displaySearch?: boolean
+  readonly searchShadow?: boolean
+  readonly displayNavButtons?: boolean
+  readonly user?: User
+}
+
+const getSearchQuery = () => {
+  const currentLocation = browserHistory.getCurrentLocation()
+
+  if (currentLocation.pathname === '/works') return (currentLocation.query as any).query
```

Comment:
This doesn't feel right. Components shouldn't access browserHistory or the query params. Can this be provided via props?

@kennylavender @wzalazar for further input

Reply:
yes, I think the same. Another problem is that the component knows about the specific name of our business like /works. I think we have to separate the responsibilities.

Reply:
OK. I kept this function from when I broke apart the Navbar component into a container and component. It was a private method on the original component.